File: DataStructure/src/main/java/com/sunandan/Random/ContainerWithMostWater.java
package com.sunandan.Random;

public class ContainerWithMostWater {

    public int maxArea(int[] height) {
        int n = height.length;
        int max = Math.min(height[0], height[n-1]) * (n-1);
        int left = 0, right = n - 1;
        while(left < right){
            max = Math.max(max, (Math.min(height[left], height[right]) * (right - left)));
            if(height[left] < height[right])
                left++;
            else
                right--;
        }
        return max;
    }

}

File: python/container_with_most_water.py
# https://leetcode.com/problems/container-with-most-water/

def container_with_most_water(height: List[int]) -> int:
	
	# Start from the widest container i.e. start and end
	# Move inward but from the one that is smaller
	# see the current water level and replace it with max or keep max

	max_water = 0
	i = 0
	j = len(height) - 1
	while i < j:
		current_water = (j - i) * min(height[i], height[j])
		max_water = max(current_water, max_water)

		if height[i] < height[j]:
			i += 1
		else:
			j -= 1

	return max_water